Feature student observation

LuanLimaOliveir opened the pull request:
Closes #104  #103

Commit:
Criada entidade StudentClassObservation

Commit:
Criado DTO StudentClassObservation

Commit:
Criado API StudentClassObservationController

Commit:
Criado repositório para salvar e consultar as observações do estudante

Commit:
Criado serviço para as observações do estudante

Commit:
Retornando DTO ao consulta das observações do estudante

Commit:
Adicionando o @RequestBody

diCaminha commented on src/main/java/com/gladguys/alucoapi/controllers/StudentClassObservationController.java:
```diff
@@ -0,0 +1,46 @@
+        Long teacherId = jwtTokenUtil.getTeacherIdFromToken(request).longValue();
+        StudentClassObservation studentClassObservation = this.studentClassObservationService.save(studentClassObservationDTO.toEntity());
+
+        return ResponseEntity.ok(studentClassObservation);
```
faz um toDTO() pra retornar

diCaminha commented on src/main/java/com/gladguys/alucoapi/entities/StudentClassObservation.java:
```diff
@@ -0,0 +1,27 @@
+
+    @JsonFormat(pattern = "dd/MM/yyyy")
+    private LocalDate date;
+    private String observation;
```
se a observação for grande, pode da problema pro tamanho do campo. usar @Column(columnDefinition = "text")

Commit:
Definindo como text o tipo da coluna observation

Commit:
Retornando DTO de observações ao persistir os dados.

LuanLimaOliveir commented:
Adicionando remoção e atualização das observações do estudante.

weslyh commented on src/main/java/com/gladguys/alucoapi/controllers/StudentClassObservationController.java:
```diff
@@ -0,0 +1,46 @@
+    @ApiOperation(value = "Retorna as observações realizadas pelo professor em sala de aula de um estudante especifico a partir do seu id")
+    @GetMapping("{studentId}")
+    public ResponseEntity<List<StudentClassObservationDTO>> get(HttpServletRequest request, @PathVariable("studentId") Long studentId) {
+        Long teacherId = jwtTokenUtil.getTeacherIdFromToken(request).longValue();
```
Nesse caso, não seria necessário utilizar esse `teacherId `?

Eu quero as observações desse `teacherId`, para esse `studentId` ... Ou não ? 

Vi que no CustomRepository ele não é utilizado.

Confesso que não sei se esse cara deveria ser utilizado... Caso seja, utiliza, se não, papoca.

Commit:
Implementado existsById - Observações do aluno

Implementado serviço que avalia a existência de uma determinada observação a partir do seu id.

Commit:
Implementado exceção - Observação do aluno não encontrada

Criada excessão a fim de informar que uma determinada observação não existe.

Commit:
Implementado deleteById - Deleção das observações dos alunos

Commit:
Implementado update - Atualização das observações dos alunos

Commit:
Implementado getById - Consultando as observações dos alunos por id
